app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +95,36 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_map()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    def _populate_min_max_map(self) -> None:
+        for answer_id, answers in self._answers_by_id.items():
+            for min_max in ["minimum", "maximum"]:
+                if (
+                    value := answers[0].get(min_max, {}).get("value")
```

Comment:
the `value := answers[0].get(min_max, {}).get("value")` needs to be in brackets - at the moment its assigning value to `True` as its setting it equal to the whole expression

If there were multiple options for an answer_id (detailed options maybe?) does this need to iterate over all of those instead of `answer[0]`?

Reply:
This code @katie-gardner suggested was already working, we probably need to revert to it and just tackle complexity by delegating to some helper method:
```python
    def _populate_min_max_map(self) -> None:
        for answer_id, answers in self._answers_by_id.items():
            if (answer_type := answers[0].get("type")) and answer_type not in [
                "Date",
                "MonthYearDate",
                "YearDate",
            ]:
                for min_max in ["minimum", "maximum"]:
                    if value := answers[0].get(min_max, {}).get("value"):
                        if isinstance(value, int):
                            self.min_and_max_map[answer_id] = str(value)
                        elif isinstance(value, dict) and value:
                            if (
                                value.get("source") == "answers"
                                and value["identifier"] in self.min_and_max_map
                            ):
                                self.min_and_max_map[answer_id] = self.min_and_max_map[
                                    value["identifier"]
                                ]
```
and then test would be:

```python
def test_schema_min_max_populate():
    schema = load_schema_from_name("test_numbers")
    assert schema.min_and_max_map == {
        "set-minimum": "1000",
        "set-maximum": "10000",
        "test-range": "10000",
        "test-range-exclusive": "10000",
        "test-min": "-123",
        "test-max": "1234",
        "test-min-exclusive": "123",
        "test-max-exclusive": "1234",
        "test-percent": "100",
        "test-decimal": "10000",
        "other-answer": "10000",
    }
```